students/MicahBraun/Lesson 06/mailroom_refactor.py

```diff
@@ -0,0 +1,334 @@
+                firstname = firstname.split(' ', ).pop(0)  # format to split on whitespace
+                first_names.append(firstname)  # append value to empty list
+
+            for i in range(int(len(list(donor_dict.values())))):  # for items in the span of donor_dict
+                elem = list(donor_dict.values())[i][0]  # elem = donor_dict at index[i], column[0]
+                elem = elem.replace(" ", "")  # format name w/o whitespace
+                names_list.append(elem)  # append to name_list
+
+            for names in range(1, int(len(names_list))):  # omit header, for each name in names_list[]
+                try:
+                    with open(cur_dir + str(names_list[names]) + '.txt',
+                              'a') as filename:  # create files/file-names
+                        filename.write(today)  # write current date (mm/dd/yy)
+                        filename.write('\n\nDear {0}, \n\nThank you for your continued support through your contribution '
+                                       'of {1} towards our Foundation\'s fundraising goal.\n\nBest wishes,\n'
+                                       'Foundation Board of Directors'
+                                       '\n'.format(str(first_names[names]),
+                                                   locale.currency(list(donor_dict.values())[names][1], grouping=True)))
+                except Exception as e:
+                    print('Error on line {}'.format(sys.exc_info()[-1].tb_lineno), type(e).__name__, e)
+            print("Files being written to " + cur_dir + "...")
+            print("Done!")
+
+        elif msg.upper() == "N":  # if "N" for msg = input()
+            input("Exiting back to menu [Press ENTER]")  # leaving this function
+            break  # back to menu()
+        else:
+            print("That's an unrecognized value! Only Y or N are accepted.")
+            continue  # warning message: back to beginning of function
+
+
+def quit_program():
+    """Function quits all processes"""
+    input("Exiting Program...[Press ENTER]")    # I use PyCharm and exit() w/o import worked fine but
+    from sys import exit                        # iPython was not running exit() without importing it from sys
+    exit()                                      # so to avoid issues, I just kept this here
+
+
+def averagedonations(donations, num):
+    """Returns average of donor's gifts"""
+    sumgifts = donations
+    numgifts = num
+
+    average_donation = sumgifts / numgifts
+    return average_donation
+
+
+'''
+#   NOTE: I could not get these 2 functions to work together... I tried breaking up the displaylist() function
+          into a processing function and a display function to better separate things for test-ability but
+          I kept running into typeErrors, inaccessible values within if-blocks, and the final attempt yielded
+          just blank returns. Not sure what to do with this section
```

Comment:
I think you need to rethink this function; the beautiful thing about using a dictionary for donors is how simple it makes to search and add information.

I am seeing so many instances of `list(donor_dict.values())` in this function, which is very in-efficient and pretty sure unnecessary. 

Remember that your dictionary should ideally store a similar like object, like holding donor information.. if you start mixing in other type of data, it makes things more challenging (I'm referring to the header).